### PR TITLE
Show ContactDrawable in chat conversation

### LINKED ISSUE #774: Show Contact Avatar in chat

### PR DESCRIPTION
**Pull request by JoGir**
Resolves #774

### CODE REVIEW

**Comment by audax** [comment]
Do we have an issue for that?

File: qabelbox/src/main/java/de/qabel/qabelbox/chat/view/adapters/ChatMessageViewHolderBase.kt [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
 
+import android.annotation.TargetApi
+import android.content.res.ColorStateList
+import android.graphics.drawable.GradientDrawable
+import android.os.Build
 import android.support.v7.widget.RecyclerView
-import android.view.LayoutInflater
 import android.view.View
+import de.qabel.chat.repository.entities.ChatDropMessage
+import de.qabel.core.ui.initials
```
Comment (audax) [inline qabelbox/src/main/java/de/qabel/qabelbox/chat/view/adapters/ChatMessageViewHolderBase.kt]: |`line 34`|
|---|
This branch is untested

**Comment by JoGir** [comment]
capture _2016-10-23-23-02-55

**Comment by audax** [comment]
Changelog entry missing 👼

File: qabelbox/src/test/java/de/qabel/qabelbox/chat/view/adapters/ShareChatMessageViewHolderTest.kt [hunk 2]
```diff
@@ -70,26 +68,38 @@ class ShareChatMessageViewHolderTest {
...
         val msg = ChatMessage(mock<Identity>(), contact,
                 direction, Date(),
-                MessagePayloadDto.ShareMessage("text", BoxFileChatShare(status, "test.txt", 2048L,
+                MessagePayloadDto.ShareMessage("text", BoxFileChatShare(status, "test.txt", fileSize,
                         SymmetricKey(listOf()), "http://foo")))
         holder.bindTo(msg, false)
 
         verify(messageField).text = "text"
-        verify(fileField).text = "test.txt 2.0KB"
+        verify(fileField).text = "test.txt " + Formatter.formatShortFileSize(RuntimeEnvironment.application, fileSize)
```
Comment (julianseeger) [inline qabelbox/src/test/java/de/qabel/qabelbox/chat/view/adapters/ShareChatMessageViewHolderTest.kt]: computation within the verification sounds like a bad Idea, how comes we no longer know the formatted filesize?

File: qabelbox/src/main/java/de/qabel/qabelbox/chat/view/adapters/ChatMessageViewHolderBase.kt (hunk 1 above)
Comment (julianseeger) [inline qabelbox/src/main/java/de/qabel/qabelbox/chat/view/adapters/ChatMessageViewHolderBase.kt]: what he said. even though the comment is not on line 34...

File: qabelbox/src/test/java/de/qabel/qabelbox/chat/view/adapters/TextChatMessageViewHolderTest.kt [hunk 3]
```diff
@@ -52,6 +67,8 @@ class TextChatMessageViewHolderTest {
 
         verify(textField).text = "text"
         verify(dateField).text = RuntimeEnvironment.application.getString(R.string.moments_ago)
+        verify(background).mutate()
```
Comment (JoGir) [inline qabelbox/src/test/java/de/qabel/qabelbox/chat/view/adapters/TextChatMessageViewHolderTest.kt]: ...

**Pull request merged**